pylinac/ct.py

```diff
@@ -1668,6 +1668,54 @@ def find_origin_slice(self) -> int:
         # drop slices that are way far from median
         hu_slices = hu_slices[((c + ln / 2) >= hu_slices) & (hu_slices >= (c - ln / 2))]
         center_hu_slice = int(round(float(np.median(hu_slices))))
+
+        # If we have a DICOM header, we'll use it to adjust how many slices we search either side. That's because the
+        # thicker the slice thickness the more we are moving in the phantom, so we quickly are beyond where the markers
+        # should be. Conversely, the thinner the slices the more slices we need to step through to cover the same
+        # physical distance. Normalised to 5 slices either side for a 2 mm scan, which seemed reasonable looking at a
+        # selection of datasets
+        if "SliceThickness" in self.dicom_stack.metadata:
+            slices_either_side = int(
+                10 / self.dicom_stack.metadata["SliceThickness"].value
+            )
+        else:
+            slices_either_side = 5
+
+        # Iterate through the slices before and after the found center of the phantom based on the original analysis,
+        # and look for the three markers at the edge of the phantom. If all three are found, store that slice. We then
+        # use these slices to 'fine tune' the central key slice locating the phantom. Note that the original analysis
+        # goes through the slices with a stride of two, this goes one by one - so we've got greater resolution during
+        # this 'fine tune' step
+        slices_where_markers_found = []
+        for image_number in range(
+            center_hu_slice - slices_either_side,
+            center_hu_slice + slices_either_side + 1,
+        ):
+            if self._is_within_image_extent(image_number):
+                slice = Slice(
+                    self, image_number, combine=False, clear_borders=self.clear_borders
+                )
+                circle_prof = CollapsedCircleProfile(
+                    center=slice.phan_center,  # This shouldn't fail as only valid slices have been selected originally
+                    radius=(self.catphan_radius_mm - 3)
+                    / self.mm_per_pixel,  # Go inside by 3 mm
+                    image_array=slice.image,
+                    start_angle=20,
+                    width_ratio=0.015,
+                    num_profiles=5,
+                )
+
+                peaks = circle_prof.find_peaks(threshold=180, min_distance=0.225)
+
+                if len(peaks[0]) == 3:
```

Comment:
The 3 markers only appear to be for the 604. The other catphans have 0 or 1 AFAICT. This means this value should probably be a class attribute so in can be controlled per catphan.

Reply:
There are similar markers in other newer CATPHAN phantoms, its just that they might be centered on other modules, some with a 4th marker on the underside. What if we had a 'refinable' property and a 'refine' method for phantoms where this technique is possible

Reply:
Ah you're right. I asked if other models had three and was told they do, but looking at some old scans we have for the 504 you're right in that it only has one. I've not sure how robust finding only one peak would be though. I'll see if I can isolate my CatPhan 504 scans and see how it looks.